cwizard2011/maintenace-tracker-app
# #158235596: implement routes for password reset

#### What does this PR do?
- Send password reset link to users email
- Validate token sent to users email
- Set new password for users

#### Description of Task to be completed?
Allows users to reset their password when they forget the password

#### How should this be manually tested?
- Clone this repository
- navigate to the ```maintenace-tracker-app```  directory
- run npm install
- create a .env file at the root of the project following the guide from ```.env.example``` file found at the root of this repository and setup your port
- run ```npm run start```
- launch postman and send a POST request to ```localhost:'your port'/api/v1/auth/passwordreset``` to login , provide your email or username as your request body, then check your registered for a confirmation link.
- Copy the id and token sent as the password reset link param from your email and send a POST request to ```localhost:'your port'/api/v1/auth/resetpassword``` and supply id, token and password in the request body to set a new password
- To run test,  run ```npm test``` in your terminal

#### What are the relevant pivotal tracker stories?
#158235596
